apps/emqx/src/emqx_metrics.erl

```diff
@@ -389,6 +464,21 @@ handle_call({set_type_to_counter, Keys}, _From, State) ->
         Keys
     ),
     {reply, ok, State};
+handle_call(#register_namespace{namespace = Namespace}, _From, State) ->
+    case persistent_term:get(?PT_KEY(Namespace), undefined) of
+        undefined ->
+            CRef = counters:new(?MAX_SIZE, [write_concurrency]),
```

Comment:
this is probably micro optimization, but we should perhaps try to limit the size for less memory consumption.
same for the default, i don't think we have 1024 counters.